extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -649,134 +607,64 @@ export class ProjectsController {
+		const profile = this.getConnectionProfileFromContext(context);
+		let createProjectFromDatabaseDialog = this.getCreateProjectFromDatabaseDialog(profile);
 
-		let profile = this.getConnectionProfileFromContext(context);
-		let connectionId, database;
-		//TODO: Prompt for new connection addition and get database information if context information isn't provided.
+		createProjectFromDatabaseDialog.createProjectFromDatabaseCallback = async (model) => await this.createProjectFromDatabaseCallback(model);
 
-		if (profile) {
-			database = profile.databaseName;
-			connectionId = profile.id;
-		}
-		else {
-			const connection = await azdata.connection.openConnectionDialog();
+		await createProjectFromDatabaseDialog.openDialog();
 
-			if (!connection) {
-				return undefined;
-			}
+		return createProjectFromDatabaseDialog;
+	}
 
-			connectionId = connection.connectionId;
+	public getCreateProjectFromDatabaseDialog(profile: azdata.IConnectionProfile | undefined): CreateProjectFromDatabaseDialog {
+		return new CreateProjectFromDatabaseDialog(profile);
+	}
 
-			// use database that was connected to
-			if (connection.options['database']) {
-				database = connection.options['database'];
-			}
-		}
+	public async createProjectFromDatabaseCallback(model: ImportDataModel) {
+		try {
+			const workspaceApi = utils.getDataWorkspaceExtensionApi();
 
-		// choose database if connection was to a server or master
-		if (!database || database === constants.master) {
-			const databaseList = await azdata.connection.listDatabases(connectionId);
-			database = (await vscode.window.showQuickPick(databaseList.map(dbName => { return { label: dbName }; }),
-				{
-					canPickMany: false,
-					placeHolder: constants.extractDatabaseSelection
-				}))?.label;
+			let newProjFolderUri = model.filePath;
 
-			if (!database) {
-				throw new Error(constants.databaseSelectionRequired);
-			}
-		}
+			const newProjFilePath = await this.createNewProject(model.projName, vscode.Uri.file(newProjFolderUri), true);
+			model.filePath = path.dirname(newProjFilePath);
+			this.setFilePath(model);
 
-		model.database = database;
-		model.serverId = connectionId;
+			const project = await Project.openProject(newProjFilePath);
+			await this.createProjectFromDatabaseApiCall(model); // Call ExtractAPI in DacFx Service
+			let fileFolderList: string[] = model.extractTarget === mssql.ExtractTarget.file ? [model.filePath] : await this.generateList(model.filePath); // Create a list of all the files and directories to be added to project
 
-		return model;
-	}
+			await project.addToProject(fileFolderList); // Add generated file structure to the project
 
-	private getConnectionProfileFromContext(context: azdata.IConnectionProfile | any): azdata.IConnectionProfile | undefined {
-		if (!context) {
-			return undefined;
+			// add project to workspace
+			workspaceApi.showProjectsView();
```

Comment:
please verify that this the Projects viewlet is focused after this dialog finishes now that the data workspace changes are merged. Thanks!

Reply:
Yes that works as expected.